src/controllers/activegate/reconciler/statefulset/statefulset_test.go

```diff
@@ -158,9 +161,9 @@ func TestStatefulSet_Volumes(t *testing.T) {
 		volumes := buildVolumes(stsProperties, getContainerBuilders(stsProperties))
 		expectedSecretName := instance.Name + "-router-" + customproperties.Suffix
 
-		require.NotEmpty(t, volumes)
+		require.Equal(t, 2, len(volumes))
 
-		customPropertiesVolume := volumes[0]
+		customPropertiesVolume := volumes[1]
```

Comment:
Please don't access slice elements by index when you mean something specific, roll out a helper function like `getVolumeByName()` or so (similar to `getContainerByName()`) or use a map.